extensions/html/server/src/test/pathCompletion/pathCompletion.test.ts

```diff
@@ -0,0 +1,125 @@
+
+	test('Adjacent Folder', () => {
+		const value = '../src/';
+		const activeFileFsPath = path.resolve(fixtureRoot, 'about/about.html');
+		const suggestions = providePathSuggestions(value, activeFileFsPath);
+
+		assert.equal(suggestions.length, 2);
+		assert.equal(suggestions[0].label, 'feature.js');
+		assert.equal(suggestions[1].label, 'test.js');
+
+		assert.equal(suggestions[0].kind, CompletionItemKind.File);
+		assert.equal(suggestions[1].kind, CompletionItemKind.File);
+	});
+
+
+});
+
+suite('Path Completion - Absolute Path', () => {
+	test('Root', () => {
+		const value = '/';
+		const activeFileFsPath1 = path.resolve(fixtureRoot, 'index.html');
+		const activeFileFsPath2 = path.resolve(fixtureRoot, 'about/index.html');
+
+		const suggestions1 = providePathSuggestions(value, activeFileFsPath1, fixtureRoot); 
+		const suggestions2 = providePathSuggestions(value, activeFileFsPath2, fixtureRoot); 
+
+		const verify = (suggestions) => {
+			assert.equal(suggestions[0].label, 'about');
+			assert.equal(suggestions[1].label, 'index.html');
+			assert.equal(suggestions[2].label, 'src');
+
+			assert.equal(suggestions[0].kind, CompletionItemKind.Folder);
+			assert.equal(suggestions[1].kind, CompletionItemKind.File);
+			assert.equal(suggestions[2].kind, CompletionItemKind.Folder);
+		};
+
+		verify(suggestions1);
+		verify(suggestions2);
+	});
+
+	test('Sub Folder', () => {
+		const value = '/src/';
+		const activeFileFsPath = path.resolve(fixtureRoot, 'about/about.html');
+		const suggestions = providePathSuggestions(value, activeFileFsPath, fixtureRoot);
+
+		assert.equal(suggestions.length, 2);
+		assert.equal(suggestions[0].label, 'feature.js');
+		assert.equal(suggestions[1].label, 'test.js');
+
+		assert.equal(suggestions[0].kind, CompletionItemKind.File);
+		assert.equal(suggestions[1].kind, CompletionItemKind.File);
+	});
+});
+
+suite('Path Completion - Incomplete Path at End', () => {
+	test('Incomplete Path that starts with slash', () => {
+		const value = '/src/f';
+		const activeFileFsPath = path.resolve(fixtureRoot, 'about/about.html');
+		const suggestions = providePathSuggestions(value, activeFileFsPath, fixtureRoot);
+
+		assert.equal(suggestions.length, 2);
+		assert.equal(suggestions[0].label, 'feature.js');
+		assert.equal(suggestions[1].label, 'test.js');
+
+		assert.equal(suggestions[0].kind, CompletionItemKind.File);
+		assert.equal(suggestions[1].kind, CompletionItemKind.File);
+	}); 
+
+	test('Incomplete Path that does not start with slash', () => {
+		const value = '../src/f';
+		const activeFileFsPath = path.resolve(fixtureRoot, 'about/about.html');
+		const suggestions = providePathSuggestions(value, activeFileFsPath, fixtureRoot);
+
+		assert.equal(suggestions.length, 2);
+		assert.equal(suggestions[0].label, 'feature.js');
+		assert.equal(suggestions[1].label, 'test.js');
+
+		assert.equal(suggestions[0].kind, CompletionItemKind.File);
+		assert.equal(suggestions[1].kind, CompletionItemKind.File);
+	}); 
+});
```

Comment:
I'd love to see a test that applies the result to the document.

Reply:
I see. I'll convert the `insertText` to `textEdit`.

Reply:
Actually I forgot -- the reason I did it was for `TextEdit`, somehow the LS client filters out all the completion items if I do not give them a `filterText`, and I don't know what meaningful `filterText` to give to the items.

Yesterday when I was testing this, this happened only on insider, not stable.
You can try running this branch to see: https://github.com/Microsoft/vscode/tree/octref/html-path-completion-textedit